File: AdminPanel/AdminPanel.Web/Modules/ECommerceWebsite/Order/OrderColumns.cs
using Serenity.ComponentModel;
using System;
using System.ComponentModel;

namespace AdminPanel.ECommerceWebsite.Columns;

[ColumnsScript("ECommerceWebsite.Order")]
[BasedOnRow(typeof(OrderRow), CheckNames = true)]
public class OrderColumns
{
    [EditLink, DisplayName("Db.Shared.RecordId"), AlignRight]
    public int Id { get; set; }
    [EditLink]
    public string Name { get; set; }
    public string Address { get; set; }
    public string Email { get; set; }
    public string Phone { get; set; }
    public string UserId { get; set; }
    public DateTime DateAdded { get; set; }
    public decimal TotalPrice { get; set; }
    public int Status { get; set; }
}

File: AdminPanel/AdminPanel.Web/Modules/ServerTypes/ECommerceWebsite/OrderColumns.ts
import { ColumnsBase, fieldsProxy } from "@serenity-is/corelib";
import { Column } from "@serenity-is/sleekgrid";
import { OrderRow } from "./OrderRow";

export interface OrderColumns {
    Id: Column<OrderRow>;
    Name: Column<OrderRow>;
    Address: Column<OrderRow>;
    Email: Column<OrderRow>;
    Phone: Column<OrderRow>;
    UserId: Column<OrderRow>;
    DateAdded: Column<OrderRow>;
    TotalPrice: Column<OrderRow>;
    Status: Column<OrderRow>;
}

export class OrderColumns extends ColumnsBase<OrderRow> {
    static readonly columnsKey = 'ECommerceWebsite.Order';
    static readonly Fields = fieldsProxy<OrderColumns>();
}